src/main/java/com/auction/auction/bid/service/AuctionService.java

```diff
@@ -91,13 +95,14 @@ public void closeAuction(String auctionId) {
         auctionRepository.save(auction);
     }
 
-
-    public Auction saveOrUpdateAuction(Auction auction) {
+    @Override
+    public Auction saveAuction(Auction auction) {
```

Comment:
This is duplicate of createAuction() method

Reply:
Hi, this is for editing an auction and when the user saves it.

Reply:
Ok, i got you. 
Generally speaking the logic  of these two methods the same, so it looks sensible to have only one method then. However these two methods should not have the same logic because when you creating an auction - you sent payload without id and the id must be generated for it, but when you update an auction - you must sent id in the payload and you must use this id to define if:
1. Auction exists
2. Auction can be updated (auction does not started) 
3. Auction has not been done (no sense to update)
4. Maybe some additional validations in future
So i just want to say that create and update logic are deferent, but in you case they are the same.
I'm okay to leave it for now like it is now (because we need soma base level of implementation) but in future you must modify update logic